config/environment.go

```diff
@@ -51,37 +52,46 @@ func applyEnvKey(key, value string, rValueSubject reflect.Value) {
 		// If we have only one part left, we can't deal with a struct
 		// the env var is incomplete so give up.
 		if len(keyParts) < 2 {
-			return
+			return false
 		}
-		applyEnvKey(keyParts[1], value, rFieldValue)
+		return applyEnvKey(keyParts[1], value, rFieldValue)
 	case reflect.String:
 		rFieldValue.Set(reflect.ValueOf(value))
+		return true
 	case reflect.Bool:
 		boolVal, err := strconv.ParseBool(value)
 		if err == nil {
 			rFieldValue.Set(reflect.ValueOf(boolVal))
+			return true
 		}
 	case reflect.Int:
 		intVal, err := strconv.ParseInt(value, 10, 0)
 		if err == nil {
 			rFieldValue.Set(reflect.ValueOf(int(intVal)))
+			return true
 		}
 	case reflect.Int64:
 		intVal, err := strconv.ParseInt(value, 10, 0)
 		if err == nil {
 			rFieldValue.Set(reflect.ValueOf(intVal))
+			return true
 		}
 	case reflect.SliceOf(reflect.TypeOf("")).Kind():
 		rFieldValue.Set(reflect.ValueOf(strings.Split(value, " ")))
+		return true
 	}
+	return false
 }
 
 func applyEnvironmentMap(inputConfig *model.Config, env map[string]string) *model.Config {
 	appliedConfig := inputConfig.Clone()
 
 	rvalConfig := reflect.ValueOf(appliedConfig).Elem()
 	for envKey, envValue := range env {
-		applyEnvKey(strings.TrimPrefix(envKey, "MM_"), envValue, rvalConfig)
+		applied := applyEnvKey(strings.TrimPrefix(envKey, "MM_"), envValue, rvalConfig)
+		if applied {
+			mlog.Debug("Mattermost is loading value from environment variable.", mlog.String("envKey", envKey), mlog.String("test-skip", ""))
```

Comment:
I think it's not ideal to have test specific code. What do you think of filtering the content in the test code? Or, simply override env vars?